cloudflare/resource_cloudflare_authenticated_origin_pulls_certificate_test.go

```diff
@@ -0,0 +1,141 @@
+package cloudflare
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccCloudflareAuthenticatedOriginPullsCertificatePerZone(t *testing.T) {
+	t.Parallel()
+	var perZoneAOP cloudflare.PerZoneAuthenticatedOriginPullsCertificateDetails
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("cloudflare_authenticated_origin_pulls_certificate.%s", rnd)
+	aopType := "per-zone"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareAuthenticatedOriginPullsCertificateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareAuthenticatedOriginPullsCertificateConfig(zoneID, rnd, aopType),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareAuthenticatedOriginPullsCertificatePerZoneExists(name, &perZoneAOP),
+					resource.TestCheckResourceAttr(
+						name, "zone_id", zoneID),
+					resource.TestCheckResourceAttr(
+						name, "type", aopType),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCloudflareAuthenticatedOriginPullsCertificatePerHostname(t *testing.T) {
+	t.Parallel()
+	var perZoneAOP cloudflare.PerHostnameAuthenticatedOriginPullsCertificateDetails
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("cloudflare_authenticated_origin_pulls_certificate.%s", rnd)
+	aopType := "per-hostname"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudflareAuthenticatedOriginPullsCertificateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareAuthenticatedOriginPullsCertificateConfig(zoneID, rnd, aopType),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudflareAuthenticatedOriginPullsCertificatePerHostnameExists(name, &perZoneAOP),
+					resource.TestCheckResourceAttr(
+						name, "zone_id", zoneID),
+					resource.TestCheckResourceAttr(
+						name, "type", aopType),
```

Comment:
Similar to above, do we need these on their own newline?

```suggestion
					resource.TestCheckResourceAttr(name, "zone_id", zoneID),
					resource.TestCheckResourceAttr(name, "type", aopType),
```